Random EnvCommandError in CI

Sometimes tests fail during the installation, like in:
```
  EnvCommandError

  Command ['/home/runner/.cache/pypoetry/virtualenvs/yapapi-VLRtOes0-py3.6/bin/pip', 'install', '--no-deps', '/home/runner/.cache/pypoetry/artifacts/9f/13/03/3682a41d00e0d034b6061b6b3d6857695e91216acac73fba2e0273b6bb/nr.stream-0.0.5-py2.py3-none-any.whl'] errored with the following return code 1, and output: 
  Processing /home/runner/.cache/pypoetry/artifacts/9f/13/03/3682a41d00e0d034b6061b6b3d6857695e91216acac73fba2e0273b6bb/nr.stream-0.0.5-py2.py3-none-any.whl
  Installing collected packages: nr.stream
  ERROR: Could not install packages due to an EnvironmentError: [Errno 2] No such file or directory: '/home/runner/.cache/pypoetry/virtualenvs/yapapi-VLRtOes0-py3.6/lib/python3.6/site-packages/nr/__init__.py'
  
  WARNING: You are using pip version 20.2.2; however, version 21.2.4 is available.
  You should consider upgrading via the '/home/runner/.cache/pypoetry/virtualenvs/yapapi-VLRtOes0-py3.6/bin/python -m pip install --upgrade pip' command.
  

  at /opt/hostedtoolcache/poetry/1.1.4/x64/lib/poetry/utils/env.py:1074 in _run
      1070│                 output = subprocess.check_output(
      1071│                     cmd, stderr=subprocess.STDOUT, **kwargs
      1072│                 )
      1073│         except CalledProcessError as e:
    → 1074│             raise EnvCommandError(e, input=input_)
      1075│ 
      1076│         return decode(output)
      1077│ 
      1078│     def execute(self, bin, *args, **kwargs):

  • Installing typing-extensions (3.10.0.2)
  • Installing zipp (3.5.0)
Error: Process completed with exit code 1.
```

This is a known problem, with some solutions, e.g. check
https://github.com/python-poetry/poetry/issues/2629

--> it would be great to apply some working solution in our github actions

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -52,7 +52,7 @@ jobs:
         id: cache
         with:
           path: .venv
-          key: venv-${{ matrix.os }}-${{ matrix.python-version }}-${{ hashFiles('**/poetry.lock') }}
+          key: venv-${{ matrix.os }}-python${{ matrix.python-version }}-${{ hashFiles('**/poetry.lock') }}
 
       - name: Ensure cache is healthy
         if: steps.cache.outputs.cache-hit == 'true'

